gradio/components/dataset.py

```diff
@@ -61,16 +63,19 @@ def __init__(
             container: If True, will place the component in a container - providing some extra padding around the border.
             scale: relative width compared to adjacent Components in a Row. For example, if Component A has scale=2, and Component B has scale=1, A will be twice as wide as B. Should be an integer.
             min_width: minimum pixel width, will wrap if not sufficient screen space to satisfy this value. If a certain scale value results in this Component being narrower than min_width, the min_width parameter will be respected first.
+            component_props: This is created when the dataset is initialized. Not intended to be set manually. But we add it to init so that Datasets can be loaded from spaces with gr.load.
+            component_ids: This is created when the dataset is initialized. Not intended to be set manually. But we add it to init so that Datasets can be loaded from spaces with gr.load.
```

Comment:
Instead of adding these parameters, which are never used, it would be better to update this logic:

https://github.com/gradio-app/gradio/blob/797e026c87983bc1b51cb3fe05e1d7b19d15635c/gradio/blocks.py#L612

to remove `component_props`

Reply:
I think we need a general pattern to exclude props that are in the config but not in the init signature. Otherwise, a custom component that sets a property in the init method that's not in the signature won't be able to be loaded. I'll think about this and push up.